src/components/MainSection.jsx

```diff
@@ -1,111 +1,117 @@
     fetch('https://65de35f3dccfcd562f5691bb.mockapi.io/api/v1/meals')
       .then((response) => response.json())
-      .then((data) => {
-        this.setState({ items: data });
-      })
-      .catch((error) => {
-        console.error("Error fetching the menu:", error);
-      });
-  }
+      .then((data) => setMenuItems(data))
+      .catch((error) => console.error("Error fetching the menu:", error));
+  }, []);
 
-  handleSeeMore = () => {
-    const { visibleItems, items } = this.state;
+  const handleSeeMore = () => {
     const newVisibleCount = visibleItems + 6;
-
-    this.setState({
-      visibleItems: newVisibleCount,
-      allItemsLoaded: newVisibleCount >= items.length,
-    });
+    setVisibleItems(newVisibleCount);
+    setAllItemsLoaded(newVisibleCount >= filteredItems.length);
   };
 
-  handleAddToCart = (itemId) => {
-    const ref = this.quantityRefs[itemId];
-    if (ref && ref.current) {
-      const quantity = Number(ref.current.value || 1);
-      if (quantity > 0) {
-        this.props.incrementItemCount(quantity);
-      }
-    } else {
-      console.error("Quantity input ref is null or undefined for item ID:", itemId);
+  const handleAddToCart = (itemId) => {
+    const ref = quantityRefs.current[itemId];
+    const quantity = Number(ref?.value || 0);
+    if (quantity > 0) {
+      incrementItemCount(quantity);
     }
   };
 
-  render() {
-    const { items, visibleItems, allItemsLoaded } = this.state;
-    const itemsToDisplay = items.slice(0, visibleItems);
+  const filteredItems = menuItems.filter(
+    (item) => item.category === filterCategory
+  );
 
-    return (
-      <main className="main-section">
-        <h1 className="main-title">Browse our menu</h1>
-        <p className="main-description">
-          Use our menu to place an order online, or{' '}
-          <span style={{ position: 'relative' }}>
-            <span className="phone" title="Phone Number: +1 (555) 123-4567">phone</span>
-          </span>{' '}
-          our store to place a pickup order. Fast and fresh food.
-        </p>
+  const itemsToDisplay = filteredItems.slice(0, visibleItems);
 
-        <div className="menu-buttons">
-          <button className="menu-button" disabled>Dinner</button>
-          <button className="menu-button" disabled>Breakfast</button>
-          <button className="menu-button" disabled>Dessert</button>
-        </div>
+  return (
+    <main className="main-section">
+      <h1 className="main-title">Browse our menu</h1>
+      <p className="main-description">
+        Use our menu to place an order online, or{' '}
+        <span style={{ position: 'relative' }}>
+          <span className="phone" title="Phone Number: +1 (555) 123-4567">phone</span>
+        </span>{' '}
+        our store to place a pickup order. Fast and fresh food.
+      </p>
+
+      <div className="menu-buttons">
+        <button
+          className={`menu-button ${filterCategory === 'Dinner' ? 'active' : ''}`}
```

Comment:
As far as you use this hardcoded word at least three times in your app, it would be better to extract it to the constant. The same for words "Dessert" and "Breakfast"